python/lsst/ts/externalscripts/auxtel/latiss_acquire_and_take_sequence.py

```diff
@@ -245,116 +285,371 @@ def set_metadata(self, metadata):
+                self.get_next_image_data_id(
+                    timeout=self.acq_exposure_time + STD_TIMEOUT
+                ),
             )
 
-            exp, dx_arcsec, dy_arcsec, peakVal = retvals[0]
-            dr_arcsec = np.sqrt(dx_arcsec ** 2.0 + dy_arcsec ** 2.0)
-            print(f"Calculated offsets [dx,dy] are [{dx_arcsec},{dy_arcsec}] arcsec")
-            print(f"Current radial pointing error is {dr_arcsec}")
+            exp = await get_image(
+                data_id,
+                datapath=self.dataPath,
+                timeout=self.acq_exposure_time + STD_TIMEOUT,
+                runBestEffortIsr=True,
+            )
 
+            # Find brightest star
+            try:
+                loop = asyncio.get_event_loop()
+                executor = concurrent.futures.ThreadPoolExecutor()
+                result = await loop.run_in_executor(executor, self.qm.run, exp)
+            except RuntimeError:
+                # FIXME: Patrick - deal with a failure to find the source here
+                pass  # and remove this
+                # Note that the source finding should be made a function
+
+            current_position = PointD(
+                result.brightestObjCentroid[0], result.brightestObjCentroid[1]
+            )
+
+            # Find offsets
+            self.log.debug(
+                f"Current brightest target position is {current_position} whereas the"
+                f"Target position is {target_position}"
+            )
+
+            dx_arcsec, dy_arcsec = calculate_xy_offsets(
+                current_position, target_position
+            )
+
+            dr_arcsec = np.sqrt(dx_arcsec ** 2 + dy_arcsec ** 2)
+
+            self.log.info(
+                f"Calculated offsets [dx,dy] are [{dx_arcsec:0.2f}, {dy_arcsec:0.2f}] arcsec as calculated"
+                f" from sequence number {data_id['seqNum']} on dayObs of {data_id['dayObs']}"
+            )
+
+            # Check if star is in place, if so then we're done
             if dr_arcsec < self.target_pointing_tolerance:
-                success = True
+                self.log.info(
+                    f"Current radial pointing error of {dr_arcsec:0.2f} arcsec is within the tolerance "
+                    f"of {self.target_pointing_tolerance} arcsec. "
+                    "Acquisition completed."
+                )
                 break
+            else:
+                self.log.info(
+                    f"Current radial pointing error of {dr_arcsec:0.2f} arcsec exceeds the tolerance"
+                    f" of {self.target_pointing_tolerance} arcsec."
+                )
 
-        if success:
-            self.log.info("Achieved centering accuracy")
-        else:
-            if (
-                iterNum != -1
-            ):  # also heinous to support max_acq_iter = 0 working TODO: fix this!
+            # Ask user if we want to apply the correction?
+            await self.checkpoint(
```

Comment:
I think this is the only checkpoint in the script... might be a good idea to add a couple more....

Reply:
Done... but i'll revisit this in the next ticket